src/components/Select/Select.jsx

```diff
@@ -180,6 +180,24 @@ export default class Select extends React.Component {
   onClick = e => {
     e.stopPropagation();
     e.nativeEvent.stopImmediatePropagation();
+    if (this.props.isSplit && !this.props.disabled) {
+       this.onButtonClick(e);
+    }
+  };
+
+  onKeyUp = e => {
+    e.stopPropagation();
+    e.nativeEvent.stopImmediatePropagation();
```

Comment:
Interesting, I guess we need this because we're mixing JS & React event listeners `e.nativeEvent.stopImmediatePropagation`

Reply:
we need this because we need to handle events propagated to parent and child selectors... and because we are manually handling this ourselves we need to stop the native and react event handlers from kicking in 😅 it does feel very clumsy though 😊

Reply:
That makes sense, thanks for the context @karelm 😄